### PR TITLE
Allow to use Final and ClassVar after python 3.13

### PR DESCRIPTION
**Pull request by aliwo**
Hi, Thanks for making a great library! I've been using mypy for years and it saved me from several (possibly happened) accidents!

## Description
this PR allows to use Final and ClassVar after python 3.13

I saw this [PR](https://github.com/python/mypy/pull/10478)

and I saw recent changes of python 3.13
https://docs.python.org/3/library/typing.html#typing.Final
image
Final now can be nested with ClassVar. so I added a version check!

### CODE REVIEW

File: mypy/typeanal.py [hunk 1]
```diff
@@ -605,7 +605,7 @@ def try_analyze_special_unbound_type(self, t: UnboundType, fullname: str) -> Typ
                     t,
                     code=codes.VALID_TYPE,
                 )
-            else:
+            elif self.options.python_version < (3, 13):
```
Comment (hauntsaninja) [inline mypy/typeanal.py:608]: We should still error in several of the cases this previously triggered, e.g. `def f(x: Final[int]) -> int: ...` (grep the test cases for `Final can be only used as an outermost qualifier` for examples)

**Review by hauntsaninja — COMMENTED** [review]
Thank you!

File: mypy/semanal.py [hunk 2]
```diff
@@ -5080,7 +5084,9 @@ def check_classvar(self, s: AssignmentStmt) -> None:
             node = lvalue.node
             if isinstance(node, Var):
                 node.is_classvar = True
-            analyzed = self.anal_type(s.type)
+            analyzed = self.anal_type(
+                s.type, allow_final_in_classvar=self.options.python_version >= (3, 13)
```
Comment (hauntsaninja) [inline mypy/semanal.py]: I think this one might be unnecessary? Let me push to your PR

**Review by hauntsaninja — APPROVED** [review]
Thank you for this improvement!

**Comment by aliwo** [comment]
@hauntsaninja Thank you very much!

**Comment by Starlightlord** [comment]
Nice catch and clean implementation. Glad to see mypy keeping pace with Python 3.13's typing enhancements. Thanks for the contribution!

**Pull request merged**